File: Sieve-of-Eratosthenes.js
var N = 200;
var primes = [];
var is_prime = [];

for (var i = 0; i < N; ++i) {
	primes.push(i);
	is_prime.push(true);
}

for (var i = 2; i < N; i++) {
	if (is_prime[i]) {
		for (var j = i+i; j < N; j += i)
			is_prime[j] = false;
	}
}

for (var i = 2; i < N; i++) {
    if (is_prime[i])
        console.log(primes[i]);
}

File: Sieve-of-Eratosthenes.cpp
#include <iostream>

int main()
{
	const unsigned int N = 200;

	unsigned int primes[200];
	bool is_prime[200];

	for (unsigned int i = 0; i < N; i++)
	{
		primes[i] = i;
		is_prime[i] = true;
	}

	for (unsigned int i = 2; i < N; i++)
	{
		if (is_prime[i])
		{
			for (unsigned int j = i * i; j < N; j += i)
				is_prime[j] = false;
		}
	}

	for (unsigned int i = 2; i < N; i++)
	{
		if (is_prime[i])
			std::cout << primes[i] << std::endl;
	}

	return 0;
}
